sdk/test/UnitTests/Tools/CustomResponses.cs

```diff
@@ -31,9 +31,9 @@ public static class CustomResponses
     {
 #if NET45
         public static void SetResponse(
-            AmazonServiceClient client, string content, string requestId, bool isOK)
+            AmazonServiceClient client, string content, string requestId, string amazonId, bool isOK)
```

Comment:
if the `amazonId` is only specific to mocking S3 requests, I would suggest giving it a more specific name like "s3ExtendedRequestId" and defaulting to null so the other tests don't need to explicitly pass in a null for this argument.

Reply:
Yes, it's an S3 specific id for internal troubleshooting. Will update.